src/main/java/com/example/hbs/service/HotelService.java

```diff
@@ -1,78 +1,105 @@
 package com.example.hbs.service;
 
+import com.example.hbs.dto.HotelRequestDto;
 import com.example.hbs.dto.HotelResponseDto;
 import com.example.hbs.enums.Role;
 import com.example.hbs.model.Hotel;
 import com.example.hbs.model.User;
 import com.example.hbs.repository.HotelRepository;
+import com.example.hbs.repository.UserRepository;
+import com.example.hbs.service.mapper.HotelListMapper;
+import com.example.hbs.service.mapper.HotelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Pageable;
-import org.springframework.http.ResponseEntity;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
 import java.util.List;
 
 @Service
 public class HotelService {
     @Autowired
     private HotelRepository hotelRepository;
 
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private HotelMapper hotelMapper;
+
+    @Autowired
+    private HotelListMapper hotelListMapper;
+
     public List<HotelResponseDto> findAllHotels(int pageNo) {
-        Pageable firstPage = PageRequest.of(pageNo-1, 20);
-        Page<Hotel> hotels = hotelRepository.findAll(firstPage);
-        List<HotelResponseDto> hotelResponseDtos = new ArrayList<>();
-        for(int iterateHotel = 0;iterateHotel<20;iterateHotel++){
-            Hotel hotel = hotels.getContent().get(iterateHotel);
-            HotelResponseDto responseDto = HotelResponseDto.builder().
-                    hotelName(hotel.getHotelName()).
-                    hotelLocation(hotel.getHotelLocation()).
-                    availableRooms(hotel.getAvailableRooms()).
-                    priceOfRoom(hotel.getPriceOfRoom()).build();
-            hotelResponseDtos.add(responseDto);
-        }
-        return hotelResponseDtos;
+        Pageable page = PageRequest.of(pageNo - 1, 20);
```

Comment:
size can be variable, 10, 20 etc. take this from params, if null, use default values as 10